src/features/courses/course-repository.ts

```diff
@@ -263,6 +263,36 @@ class CourseRepository {
         }
     }
 
+    async extendTopicByUser(options: ExtendTopicForUserOptions): Promise<UpsertResult<StudentTopicOverride>> {
```

Comment:
NIT: I prefer if, else here
When I first read it I was going to comment it should be in a transaction since two sql commands are run (create and update)
After looking a little harder I noticed that the first if returned
I prefer being explicit because if you ever did want some code to run regardless then you could just add it after the if, else block

Reply:
That's just my preference though ^ no need to change it if you don't want